stdlib/runtime/string.gr

```diff
@@ -102,31 +102,57 @@ let getVariantName = variant => {
         Memory.incRef(WasmI32.fromGrain(_ERR))
       }
     },
-    _ => {
-      let mut block = findTypeMetadata(typeId)
+    _ => -1n,
+  }
+}
 
-      if (block == -1n) {
-        -1n
-      } else {
-        let sectionLength = WasmI32.load(block, 0n)
-        block += 8n
-
-        let end = block + sectionLength
-        let mut result = -1n
-        while (block < end) {
-          if (WasmI32.load(block, 4n) == variantId) {
-            let length = WasmI32.load(block, 8n)
-            let str = allocateString(length)
-            Memory.copy(str + 8n, block + 12n, length)
-            result = str
-            break
-          }
-          block += WasmI32.load(block, 0n)
-        }
+@unsafe
+let getFieldArray = (fields, arity) => {
+  let fieldArray = allocateArray(arity)
+
+  let mut fieldOffset = 0n
+  for (let mut i = 0n; i < arity; i += 1n) {
+    let fieldLength = WasmI32.load(fields + fieldOffset, 4n)
+    let fieldName = allocateString(fieldLength)
+    Memory.incRef(fieldName)
+    Memory.copy(fieldName + 8n, fields + fieldOffset + 8n, fieldLength)
+    WasmI32.store(fieldArray + i * 4n, fieldName, 8n)
+
+    fieldOffset += WasmI32.load(fields + fieldOffset, 0n)
+  }
 
-        result
+  fieldArray
+}
+
+@unsafe
+let getVariantMetadata = variant => {
+  let typeId = WasmI32.load(variant, 8n) >> 1n
+  let variantId = WasmI32.load(variant, 12n) >> 1n
+
+  let mut block = findTypeMetadata(typeId)
+
+  if (block == -1n) {
+    return {
```

Comment:
It seems that this did not work without the curly braces here (though `return block` works fine just a few lines below). Is this a bug?

Reply:
Yes, this is a bug.